File: BinaryTree/java/depthFirstValues.java
import java.util.ArrayList;
import java.util.List;
import java.util.Stack;

class DepthFirstValues {
  public static List<Integer> depthFirstValues(TreeNode root) {
    List<Integer> values = new ArrayList<>();
    if (root == null) {
      return values;
    }

    Stack<TreeNode> stack = new Stack<>();
    stack.push(root);
    while (!stack.isEmpty()) {
      TreeNode current = stack.pop();
      values.add(current.val);
      if (current.right != null) stack.push(current.right);
      if (current.left != null) stack.push(current.left);
    }

    return values;
  }

  public static void run() {
    // intentionally empty
  }
}

File: BinaryTree/csharp/DepthFirstValues.cs
using System.Collections.Generic;

namespace BinaryTreeSolutions;

public static class DepthFirstValues
{
    public static IList<int> Solve(TreeNode? root)
    {
        var values = new List<int>();
        if (root is null)
        {
            return values;
        }

        var stack = new Stack<TreeNode>();
        stack.Push(root);
        while (stack.Count > 0)
        {
            var current = stack.Pop();
            values.Add(current.Val);
            if (current.Right is not null) stack.Push(current.Right);
            if (current.Left is not null) stack.Push(current.Left);
        }

        return values;
    }
}
